src/traces/bar/plot.js

```diff
@@ -231,7 +235,7 @@ function appendBarText(gd, plotinfo, bar, calcTrace, i, x0, x1, y0, y1, opts) {
 
     if(!text ||
         textPosition === 'none' ||
-        (calcBar.isBlank && (
+        ((calcBar.isBlank || x0 === x1 || y0 === y1) && (
```

Comment:
So here we don't show text associated with zero-height bars?

Reply:
Oh sorry, we don't show **inside** text associated with zero-height bars. Nicely done :ok_hand: